src/cloudai/configurator/cloudai_gym.py

```
@@ -170,19 +293,46 @@ def step(self, action: Any) -> Tuple[list, float, bool, dict]:
+        return observation, reward, done, info
+
     def render(self, mode: str = "human"):
         """
         Render the current state of the TestRun.
 
         Args:
             mode (str): The mode to render with. Default is "human".
         """
-        print(f"Step {self.test_run.current_iteration}: Parameters {self.test_run.test.cmd_args}")
+        if self._is_online:
+            logging.info(f"CloudAIGymEnv [online] step {self._online_step_count}")
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)